src/ida_name.py

```diff
@@ -0,0 +1,239 @@
+# @category GCL
+
+import cp
+from ghidra.program.flatapi import FlatProgramAPI
+from ghidra.program.model.symbol import SourceType
+from ghidra.util.exception import DuplicateNameException, InvalidInputException
+from ghidra.program.model.address import Address
+
+# SN_ Constants (set_name() flags)
+SN_CHECK = 0x00  # Fail if the name contains invalid characters.
+SN_NOCHECK = 0x01  # Don't fail, replace invalid chars with '_'
+SN_PUBLIC = 0x02  # Make name public
+SN_NON_PUBLIC = 0x04  # Make name non-public
+SN_WEAK = 0x08  # Make name weak
+SN_NON_WEAK = 0x10  # Make name non-weak
+SN_AUTO = 0x20  # Mark name as autogenerated
+SN_NON_AUTO = 0x40  # Mark name as non-autogenerated
+SN_NOLIST = 0x80  # Exclude from the name list
+SN_NOWARN = 0x100  # Do not display a warning if it fails
+SN_LOCAL = 0x200  # Create a local name
+SN_IDBENC = (
+    0x400  # Name is in IDB encoding; non-ASCII bytes will be decoded accordingly
+)
+SN_FORCE = 0x800  # If the name already exists, append a numeric suffix like "_123"
+SN_NODUMMY = 0x1000  # Prepend '_' if the name starts with a dummy prefix like 'sub_'
+SN_DELTAIL = (
+    0x2000  # If name cannot be set due to a tail byte, delete the hindering item
+)
+
+# GN_ Constants (get_name() flags)
+GN_VISIBLE = 0x0001  # Replace forbidden characters by SUBSTCHAR
```

Comment:
likewise above.

By using a (e.g. Flag) subtype, you ensure that flags are not wrongly combiner (e.g. OR-ed).